### PR TITLE
Change documentation link to new /latest URL and add robots.txt to kick out archive and 1_14 documentation folder

### PR DESCRIPTION
**Pull request by loricooperhdf**

### CODE REVIEW

**Review by lkurz — APPROVED** [review]

**Comment by lkurz** [comment]
@loricooperhdf @bmribler @byrnHDF 
I made a hand edit to the page at SUPPORT bucket to get the latest link online.  
Update the PR with the change(s) let's get it reviewed and merged soon.  
ALSO until we get past these toddler steps can we all make sure we have a "All Activity" watch set on this repo.

File: documentation/index.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ redirect_from:
 # HDF Software Documentation
 
 ## HDF5 Library, Tools, and Extensions
-* [Documentation (latest)](https://support.hdfgroup.org/releases/hdf5/latest-docs.html) - User Guide, Reference Manual, and other documentation
+* [Documentation (latest)](/hdf5/latest/) - User Guide, Reference Manual, and other documentation
```
[the remark this answers is not in the document]
Comment (loricooperhdf) [reply documentation/index.md:10]: But we're already in the documentation directory? Right?

File: robots.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+robots.txt generated by www.seoptimer.com
+User-agent: *
+Disallow: 
+Disallow: /cgi-bin/
+Disallow: /archive/
+Disallow: /releases/hdf5/v1_14/
+Sitemap: https://support.hdfgroup.org/sitemap.xml
```
[the remark this answers is not in the document]
Comment (loricooperhdf) [reply robots.txt:7]: @lkurz see internal discussion re sitemap at https://hdfgroup.atlassian.net/issues/IWIP-513

let's run with this robots and add ftp or subdirectories later. I don't have insight into that directory (sitemap would help me, too)

**Review by loricooperhdf — COMMENTED** [review]
Lori and Les conversation resolved. Needs another reviewer.

File: documentation/index.md
Comment (lkurz) [inline documentation/index.md:1]: HDFVIEW and HDF5/JSON links can also be changed to latest:

* [User Guide (lastest)](/documentation/hdfview/latest/) - User Guide
* [User Guide (latest)](/documentation/hdf5-json/latest/)

Comment (bmribler) [reply documentation/index.md:1]: @loricooperhdf Do these need to be fixed?  I'll approve it though.

**Review by bmribler — APPROVED** [review]

File: documentation/index.md
Comment (loricooperhdf) [reply documentation/index.md:1]: Made those changes.

**Pull request merged**